src/node_worker.cc

```diff
@@ -361,10 +361,19 @@ void Worker::StartThread(const FunctionCallbackInfo<Value>& args) {
 
   w->env()->add_sub_worker_context(w);
   w->stopped_ = false;
+  w->thread_joined_ = false;
+
+  w->thread_exit_async_.reset(new uv_async_t);
```

Comment:
Question: why is this before the `uv_thread_create`? Maybe it's just a non intuitive name `OnThreadStopped`?
AFAICT `uv_async_send(thread_exit_async_.get())` can only run iff the thread was created.

Reply:
> Question: why is this before the `uv_thread_create`?

Switching these would create a race condition – the worker thread might finish (and try to call `uv_async_send()` on this) before we have initialized the async

> Maybe it's just a non intuitive name `OnThreadStopped`?

Can you expand on this? It’s an event handler that’s called when the worker thread finishes.

> AFAICT `uv_async_send(thread_exit_async_.get())` can only run iff the thread was created.

That is correct, yes.